src/components/models/CartModel.ts

```diff
@@ -0,0 +1,232 @@
+            product.inCart -= 1;
+            this.productsQuantity -= 1;
+            this.totalSum = Number((this.totalSum -= product.price).toFixed(2));
+            product.sum = Number((product.sum -= product.price).toFixed(2));
+            this.calculateAddingSumDiscoutDelete();
+            this.appController.cartView.updateCartInfo();
+            this.appController.cartView.updatePromoBlock();
+            this.appController.cartView.updatePage();
+            this.appController.cartView.updatePromoBlock();
+            this.appController.cartView.updateDiscountInfo();
+        }
+    }
+
+    private calculateAddingSumDiscoutDelete() {
+        const currentTotalPercentDiscount = this.getCurrentTotalPercentDiscount();
+        this.totalSumWithDiscount = Number((this.totalSum * (1 - currentTotalPercentDiscount / 100)).toFixed(2));
+    }
+
+    public checkProductInCart(id: number) {
+        const idProductInCart = this.productsInCart.find((product) => product.id === id);
+        if (idProductInCart) {
+            return this.productsInCart.includes(idProductInCart);
+        } else {
+            return false;
+        }
+    }
+
+    public findEnteredPromoInPromocodes(enteredValue: string) {
+        this.promocodes.forEach((promocode) => {
+            if (enteredValue.toLocaleUpperCase() === promocode.value) {
+                if (!this.enteredPromocodes.find((enteredPromocode) => enteredPromocode.id === promocode.id)) {
+                    this.enteredPromocodes.push(promocode);
+                    this.appController.cartView.createPromoCodeView(promocode);
+                }
+            }
+        });
+    }
+
+    public handleAddPromocode(promocode: Promocode) {
+        if (this.productsInCart.length !== 0) {
+            if (!this.activatedPromocodes.find((activatedPromocode) => activatedPromocode.id === promocode.id)) {
+                this.activatedPromocodes.push(promocode);
+                promocode.active = true;
+                this.totalSumWithDiscount = Number(
+                    (this.totalSumWithDiscount -= this.totalSum * (promocode.discount / 100)).toFixed(2)
+                );
+                this.appController.cartView.createBlockWithDiscountSum();
+            }
+        }
+    }
+
+    public handleDeletePromocode(promocode: Promocode) {
+        if (this.productsInCart.length !== 0) {
+            const deletedPromocodeId = this.activatedPromocodes.indexOf(promocode);
+            promocode.active = false;
+            this.activatedPromocodes.splice(deletedPromocodeId, 1);
+            this.totalSumWithDiscount = Number(
+                (this.totalSumWithDiscount += this.totalSum * (promocode.discount / 100)).toFixed(2)
+            );
+            this.appController.cartView.createBlockWithDiscountSum();
+        }
+    }
+
+    public checkIsPromocodesAplied() {
+        if (this.totalSum !== this.totalSumWithDiscount) {
+            return true;
+        }
+        return false;
```

Comment:
I suggest you to replace "if (this.totalSum !== this.totalSumWithDiscount) {
            return true;
        }
        return false;" with the "return this.totalSum !== this.totalSumWithDiscount"

Reply:
Fix it